Rubberduck.Inspections/Concrete/UnreachableCaseInspection.cs

```diff
@@ -0,0 +1,1309 @@
+                    Ranges = new List<Tuple<UnreachableCaseInspectionValue, UnreachableCaseInspectionValue>>(),
+                    RangeClausesAsText = new List<string>(),
+                    EnumDiscretes = new List<UnreachableCaseInspectionValue>()
+                };
+                CanBeInspected = TryGetChildContext(SelectStmtContext, out SelectExpressionContext);
+            }
+        }
+
+        internal struct CaseClauseDataObject
+        {
+            public ParserRuleContext CaseContext;
+            public List<RangeClauseDataObject> RangeClauseDOs;
+            public ClauseEvaluationResult ResultType;
+
+            public CaseClauseDataObject(ParserRuleContext caseClause)
+            {
+                CaseContext = caseClause;
+                RangeClauseDOs = new List<RangeClauseDataObject>();
+                ResultType = ClauseEvaluationResult.NoResult;
+            }
+        }
+
+        internal struct RangeClauseDataObject
+        {
+            public RangeClauseContext Context;
+            public bool UsesIsClause;
+            public bool IsValueRange;
+            public bool IsConstant;
+            public bool CompareByTextOnly;
+            public string IdReferenceName;
+            public string AsText;
+            public string TypeNameTarget;
+            public string CompareSymbol;
+            public UnreachableCaseInspectionValue SingleValue;
+            public UnreachableCaseInspectionValue MinValue;
+            public UnreachableCaseInspectionValue MaxValue;
+            public ClauseEvaluationResult ResultType;
+            public bool CanBeInspected;
+
+            public RangeClauseDataObject(RangeClauseContext ctxt, string targetTypeName)
+            {
+                Context = ctxt;
+                UsesIsClause = false;
+                IsValueRange = false;
+                IsConstant = false;
+                CanBeInspected = true;
+                CompareByTextOnly = false;
+                IdReferenceName = string.Empty;
+                AsText = ctxt.GetText();
+                TypeNameTarget = targetTypeName;
+                CompareSymbol = CompareTokens.EQ;
+                SingleValue = null;
+                MinValue = null;
+                MaxValue = null;
+                ResultType = ClauseEvaluationResult.NoResult;
+            }
+        }
+
+        public UnreachableCaseInspection(RubberduckParserState state)
+            : base(state, CodeInspectionSeverity.Suggestion) { }
+
+        public override IInspectionListener Listener { get; } =
+            new UnreachableCaseInspectionListener();
+
+        public override CodeInspectionType InspectionType => CodeInspectionType.CodeQualityIssues;
+
+        private static Dictionary<string, string> CompareInversions = new Dictionary<string, string>()
+        {
+            { CompareTokens.EQ, CompareTokens.EQ },
+            { CompareTokens.NEQ, CompareTokens.NEQ },
```

Comment:
why is NEQ the inverse of NEQ and EQ the inverse of EQ? Shouldn't it be:
```
{ CompareTokens.EQ, CompareTokens.NEQ },
{ CompareTokens.NEQ, CompareTokens.EQ },
```

Reply:
Yeah...I think a better name for the dictionary might be "AlgebraicLogicInversions".  The goal is to be able to perform some (very) limited algebra for logical Range Clause expressions like '6 < x'.  In order to move 'x' to the LHS, the desired results is x > 6.  So, for '6 = x' the algebra results desired is 'x = 6'.  Same for NEQ.  I'm definitely open to a better name.  Note: I am interested in moving the variable to the LHS so that the expression can be evaluated in the same way as an IS clause. ('x < 6' == 'Is < 6' when 'x' is the Select Case expression)